streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -256,8 +258,7 @@ private boolean setState(final State newState) {
                 // will be refused but we do not throw exception here, to allow idempotent close calls
                 return false;
             } else if (!state.isValidTransition(newState)) {
-                log.error("Unexpected state transition from {} to {}", oldState, newState);
-                throw new IllegalStateException(logPrefix + "Unexpected state transition from " + oldState + " to " + newState);
+                throw new IllegalStateException("stream-client " + clientId + " Unexpected state transition from " + oldState + " to " + newState);
```

Comment:
Nit: 
`throw new IllegalStateException("Stream-client " + clientId + ": Unexpected state transition from " + oldState + " to " + newState);`

Capitalize `S` and add `:` (same blow)